src/App.vue

```diff
@@ -338,14 +339,27 @@ export default {
       );
       window.localStorage.setItem(nextKey, JSON.stringify(this.layoutState));
 
-      window.localStorage.removeItem(constants.LAYOUT_STATE_KEY);
-
-      window.localStorage.setItem(
-        constants.LAYOUT_LOAD_ERROR_KEY,
-        JSON.stringify(true)
+      console.warn(
+        "Layout could not be restored. Default layout loaded and old layout was saved to a backup local storage key"
       );
 
-      window.location.reload();
+      this.resetGoldenLayoutState();
+      this.restartLayout();
+    },
+
+    resetGoldenLayoutState() {
+      window.localStorage.removeItem(constants.LAYOUT_STATE_KEY);
+      this.layoutState = undefined;
+    },
+
+    /**
+     * @description Restarts Golden Layout.
+     * We increment a variable which is assigned to the key of the root Golden Layout element.
+     * If the key is updated, the element is forced to dismount and mount again.
+     */
+    restartLayout() {
+      this.resetGoldenLayoutState();
+      this.triggerUiRestart++;
```

Comment:
Why is this needed?

Reply:
> I have one question related to the code and one questoin regarding this warning that comes up when you reset the layout:
> 
> `Did not remove inputLink. Could not find input with id undefined`.
> 
> Why is `removeInputLink` called at all when we do this? Is the layout somehow also connected to an `inputLink`?

Yes, the layout contains the Gallery. Each Gallery Item is an active Module, so when the layout resets the gallery must unmount the Gallery Items and also clear up the active Modules, which does a check for InputLinks. As the Gallery Item active Modules are special Gallery ones, they have no inputs and we get these warnings.

We should add a case to check if the active Module is a module in the gallery and suppress these warnings.

Reply:
>Why is this needed?

In Vue we can't programmatically unmount a component without using a `v-if` which unfortunately removes the elements from the DOM. We can force the component to re-mount without changing the DOM using `:key`, something we need to do to make Golden Layout initialise again with the updated layout state.